pkg/scheduler/scheduler.go

```diff
@@ -249,7 +249,7 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 
 		if features.Enabled(features.TASFailedNodeReplacementFailFast) && workload.HasTopologyAssignmentWithUnhealthyNode(e.Obj) && mode != flavorassigner.Fit {
 			// evict workload we couldn't find the replacement for
-			if err := s.evictWorkloadAfterFailedTASReplacement(ctx, log, e.Obj.DeepCopy()); err != nil {
+			if err := s.evictWorkloadAfterFailedTASReplacement(ctx, log, e.Obj.DeepCopy()); client.IgnoreNotFound(err) != nil {
```

Comment:
Nice, I think this should work indeed, because we will later skip calling requeueAndUpdate on e.status == evicted. Still, I'm not clear how the test is proving that.

Reply:
Regarding testing: I would much rather see in the testCase options a new param like `wantSecondPassQueueSize` and assert `wantSecondPassQueueSize==0` after the test cases.